### ISSUE TITLE
One pass is not enough to for auto correction to fix all issues

### ISSUE THREAD

**Issue by MyDogTom** [shyiko/ktlint#145]
Sometimes, KtLint is not able to auto correct code when several violations are present. 

**Example**

Initial code
```kotlin
fun funcA(a: String): String
    = funcB(a)
```
There are two violations: `=` should be on same line and continuation indent should be used (6 in my project)

After first run
```kotlin
fun funcA(a: String): String =
    funcB(a)
```

After second run
```kotlin
fun funcA(a: String): String =
      funcB(a)

```

**Why this happens**
In `StandardRuleSetProvider` `IndentationRule` is declared before `NoLineBreakBeforeAssignmentRule`. During first run `IndentationRule` doesn't detekt anything, `NoLineBreakBeforeAssignmentRule`  changes code and after that `IndentationRule` is capable to find violation.

**Proposals**
1. During autocorrection run all rules multiple times until there is no code correction occurs. Limit maximum number of iterations to 5 or 10, in order to prevent infinite loop.

2. Try to fix that case and similar by tweaking rule order. In that case, `NoLineBreakBeforeAssignmentRule` should be before `IndentationRule`. I'm not sure if all cases can be solved like that.

**Comment by shyiko** [comment]
Partially addressed in https://github.com/shyiko/ktlint/commit/6806c1c425c57687d07c69dff40c9fd56bf63850#diff-25d2db63b28b3096a8738c2116568582. While order of the rules can still influence the outcome they are no longer run in parallel which (together with [Rule.Modifier](https://github.com/shyiko/ktlint/blob/master/ktlint-core/src/main/kotlin/com/github/shyiko/ktlint/core/Rule.kt#L48)s) should minimize if not eliminate any potential conflicts.
